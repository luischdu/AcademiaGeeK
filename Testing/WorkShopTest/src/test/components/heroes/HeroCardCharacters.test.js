import React from 'react'
import { shallow } from 'enzyme'
import { HeroCard } from '../../../components/heroes/HeroCard'
import '../../../setupTests'

describe('Pruebas con HeroCardCharacters', () => {

    test('Validar personajes de heroe', () => {
        const id = 'dc-flash'
        const superhero = 'Flash'
        const alter_ego = 'Jay Garrick'
        const first_appearance = 'Flash Comics #1'
        const characters = 'Jay Garrick, Barry Allen, Wally West, Bart Allen'

        const wrapper = shallow(
            <HeroCard
                id={id}
                superhero={superhero}
                alter_ego={alter_ego}
                first_appearance={first_appearance}
                characters={characters} />
        )
        const textParrafo = wrapper.find('.card-text-characters').text().trim()
        expect(wrapper).toMatchSnapshot()
        expect(textParrafo).toBe("Jay Garrick, Barry Allen, Wally West, Bart Allen")
        expect(wrapper.exists('.card-text-characters')).toBe(true)
    })

    test("Validar si el personaje es igual al caracter", () => {
        // const id = 'dc-batman'
        // const superhero = 'Batman'
        const alter_ego = 'Bruce Wayne'
        // const first_appearance = 'Detective Comics #27'
        const characters = 'Bruce Wayne'

        // const wrapper = shallow(
        //     <HeroCard
        //         id={id}
        //         superhero={superhero}
        //         publisher={publisher}
        //         alter_ego={alter_ego}
        //         first_appearance={first_appearance}
        //         characters={characters} />
        // )
        expect(characters).toBe(alter_ego)
    })

})