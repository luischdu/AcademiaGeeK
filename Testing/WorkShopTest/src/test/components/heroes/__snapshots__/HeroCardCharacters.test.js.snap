// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con HeroCardCharacters Validar personajes de heroe 1`] = `
<div
  className="card ms-3 animate__animated animate__fadeIn"
  style={
    Object {
      "maxWidth": 540,
    }
  }
>
  <div
    className="row no-gutters"
  >
    <div
      className="col-md-4"
    >
      <img
        alt="Flash"
        className="card-img"
        src="./assets/heroes/dc-flash.jpg"
      />
    </div>
    <div
      className="col-md-8"
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
           
          Flash
           
        </h5>
        <p
          className="card-text"
        >
           
          Jay Garrick
           
        </p>
        <p
          className="card-text-characters"
        >
           
          Jay Garrick, Barry Allen, Wally West, Bart Allen
           
        </p>
        <p
          className="card-text"
        >
          <small
            className="text-muted"
          >
             
            Flash Comics #1
             
          </small>
        </p>
        <Link
          to="./hero/dc-flash"
        >
          MÃ¡s...
        </Link>
      </div>
    </div>
  </div>
</div>
`;
